tests/components/freebox/test_config_flow.py

```diff
@@ -96,53 +76,55 @@ async def test_zeroconf(hass):
     assert result["step_id"] == "link"
 
 
-async def test_link(hass, connect):
+async def test_link(hass: HomeAssistantType, router: Mock):
     """Test linking."""
     result = await hass.config_entries.flow.async_init(
```

Comment:
You need to patch async_setup and async_setup_entry, because the result is `RESULT_TYPE_CREATE_ENTRY`.

Reply:
This is normal for this integration.

The fist step ("user") is asking the user the host and port of the router.
The second and last step ("link") prevent to user that he should push a button on the router to authenticate.
When the user submit : the integration is waiting for the user to press the button, and then, creates the config entry and the entity entries.

As `hue` as I can remember.

Reply:
Since you don't actually care about setting up the integration, you should mock out the setup methods. You only care about the config flow.

Reply:
OK, understood 😊